webapp/src/components/content/textElement.tsx

```diff
@@ -11,26 +13,56 @@ import {MarkdownEditor} from '../markdownEditor'
 
 import {contentRegistry} from './contentRegistry'
 
+import './textElement.scss'
+
 type Props = {
     block: ContentBlock
     readonly: boolean
 }
 
-const TextElement = (props: Props): JSX.Element => {
-    const {block, readonly} = props
+const BlockTitleMaxBytes = 65535 // Maximum size of a TEXT column in MySQL
+const BlockTitleMaxRunes = BlockTitleMaxBytes / 4 // Assume a worst-case representation
+
+const TextElement = ({block, readonly}: Props): JSX.Element => {
     const intl = useIntl()
 
+    const [isError, setIsError] = useState<boolean>(false)
+    const [blockTitle, setBlockTitle] = useState(block.title)
+
+    const textChangedHandler = (text: string): void => {
+        setBlockTitle(text)
+        const textSize = new Blob([text]).size
```

Comment:
I'm pretty sure we just check `text.length`. I'm not 100% sure, but that seems to account for unicode characters taking up more than one byte
![Screenshot 2024-06-13 at 5 48 29 PM](https://github.com/mattermost/focalboard/assets/3277310/93f5d47e-fb6f-47d7-816c-e1082bae8594)

Reply:
I suppose we can use `text.length` directly. Initially, I thought this would be for file uploads like images on the block, but file uploads use different APIs. I'll make those changes. Nice catch 🙌.